### PR TITLE
*: optimize auth/etcdserver logs to facilitate troubleshooting data inconsistency

### PR DESCRIPTION
**Pull request by tangcong**
There are three minor changes：
1. add warning log for troubleshooting when auth revision is inconsistent(current,no log is printed when node failed to apply command because of inconsistent auth revision).
for example.
`
17:04:56 etcd1 | {"level":"warn","ts":"2020-03-03T17:04:56.173+0800","caller":"auth/store.go:856","msg":"request auth revision is less than current node auth revision","auth store revision":23,"request auth revision":19,"request key":"hello","error":"auth: revision in header is old"}
`

2. no need to save consistentIndex in NewAuthStore(no command is executed) and  it will cause the error log to be printed when starting an empty etcd cluster.

3. print warning log when failed to apply request.

### CODE REVIEW

File: auth/store.go [hunk 1]
```diff
@@ -851,8 +851,9 @@ func (as *authStore) isOpPermitted(userName string, revision uint64, key, rangeE
 	if revision == 0 {
 		return ErrUserEmpty
 	}
-
-	if revision < as.Revision() {
+	rev := as.Revision()
+	if revision < rev {
+		as.lg.Warn("isOppermitted", zap.Uint64("auth store revision", rev), zap.Uint64("request auth revision", revision), zap.ByteString("request key", key), zap.Error(ErrAuthOldRevision))
```
Comment (jingyih) [inline auth/store.go]: Change the error message "isOppermitted" to something more relevant? Maybe "operation on key is not permitted". The logger already prints line number, so we know it is from function isOpPermitted. Also, I think multiline is preferred for better readability.

Comment (tangcong) [reply auth/store.go]: thanks. updated. PTAL.

**Comment by tangcong** [comment]
> The content LGTM. The commit logs should be `auth: ...`. etcd's commit log format is `<package name>: <a short description of the commit>`.

**Comment by tangcong** [comment]
@mitake thanks. you are careful. updated.

**Comment by tangcong** [comment]
we encounter another serious data inconsistency issue and can make sure that it has nothing to do with auth revision. we are striving to reproduce it but there is no valid information in etcd log, so it is better to print error log when failed to apply request. i have added it in this pr. 
@jingyih @mitake PTAL. thanks.

File: etcdserver/apply.go [hunk 2]
```diff
@@ -132,6 +132,9 @@ func (a *applierV3backend) Apply(r *pb.InternalRaftRequest) *applyResult {
 	ar := &applyResult{}
 	defer func(start time.Time) {
 		warnOfExpensiveRequest(a.s.getLogger(), start, &pb.InternalRaftStringer{Request: r}, ar.resp, ar.err)
+		if ar.err != nil {
```
Comment (jingyih) [inline etcdserver/apply.go]: Wondering how much does this affect server throughput. Maybe use debug level to print?

**Comment by jingyih** [comment]
> we encounter another serious data inconsistency issue and can make sure that it has nothing to do with auth revision. we are striving to reproduce it but there is no valid information in etcd log, so it is better to print warning log when failed to apply request. i have added it in this pr.
> @jingyih @mitake PTAL. thanks.

FYI, there is a known data inconsistency bug which is fixed by https://github.com/etcd-io/etcd/pull/11613.

File: auth/store.go [hunk 3]
```diff
@@ -1070,7 +1070,6 @@ func NewAuthStore(lg *zap.Logger, be backend.Backend, tp TokenProvider, bcryptCo
 
 	if as.Revision() == 0 {
 		as.commitRevision(tx)
-		as.saveConsistentIndex(tx)
```
Comment (mitake) [inline auth/store.go]: Just want to make sure the motivation of dropping this line. `NewAuthStore()` isn't called as a command apply so you are dropping it, right?

File: etcdserver/apply.go (hunk 2 above)
Comment (mitake) [reply etcdserver/apply.go]: The logging overhead will be introduced only for failed requests, so I feel adding the warning log would help troubleshooting.

File: auth/store.go (hunk 3 above)
Comment (tangcong) [reply auth/store.go]: yes, it will also cause the error log to be printed because syncConsitentIndexFunc is nil at this time.

Comment (mitake) [reply auth/store.go]: Got it, then it's fine :+1:

File: etcdserver/apply.go (hunk 2 above)
Comment (tangcong) [reply etcdserver/apply.go]: yes ,i agree with @mitake. if there are too much failed log,something may be wrong with the cluster and users can also adjust the log leve to level. @jingyih

**Comment by tangcong** [comment]
> > we encounter another serious data inconsistency issue and can make sure that it has nothing to do with auth revision. we are striving to reproduce it but there is no valid information in etcd log, so it is better to print warning log when failed to apply request. i have added it in this pr.
> > @jingyih @mitake PTAL. thanks.
> 
> FYI, there is a known data inconsistency bug which is fixed by #11613.

We have ruled out this error and have reproduced it, and we are investigating further.

**Comment by tangcong** [comment]
#11689 @jingyih 
> > we encounter another serious data inconsistency issue and can make sure that it has nothing to do with auth revision. we are striving to reproduce it but there is no valid information in etcd log, so it is better to print warning log when failed to apply request. i have added it in this pr.
> > @jingyih @mitake PTAL. thanks.
> 
> FYI, there is a known data inconsistency bug which is fixed by #11613.

#11689

**Pull request merged**